File: resources/expected/ros2/datatype-examples/strict/src/datatypes_system_cpp_pkg/src/user_code/consumer_consumer_src.cpp
#include "datatypes_system_cpp_pkg/user_headers/consumer_consumer_src.hpp"
#include "datatypes_system_cpp_pkg/base_headers/enum_converter.hpp"

//===========================================================
// This file will not be overwritten when re-running Codegen
//===========================================================

//=================================================
//  I n i t i a l i z e    E n t r y    P o i n t
//=================================================
void consumer_consumer::initialize()
{
    PRINT_INFO("Initialize Entry Point invoked");

    // Initialize the node
}

//=================================================
//  C o m p u t e    E n t r y    P o i n t
//=================================================
void consumer_consumer::handle_myBoolean(const datatypes_system_cpp_pkg_interfaces::msg::Boolean msg)
{
    // Handle myBoolean msg
    PRINT_INFO("Received myBoolean: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myInteger(const datatypes_system_cpp_pkg_interfaces::msg::Integer64 msg)
{
    // Handle myInteger msg
    PRINT_INFO("Received myInteger: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myFloat(const datatypes_system_cpp_pkg_interfaces::msg::Float64 msg)
{
    // Handle myFloat msg
    PRINT_INFO("Received myFloat: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myCharacter(const datatypes_system_cpp_pkg_interfaces::msg::Character msg)
{
    // Handle myCharacter msg
    PRINT_INFO("Received myCharacter: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myString(const datatypes_system_cpp_pkg_interfaces::msg::String msg)
{
    // Handle myString msg
    PRINT_INFO("Received myString: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myInt8(const datatypes_system_cpp_pkg_interfaces::msg::Integer8 msg)
{
    // Handle myInt8 msg
    PRINT_INFO("Received myInt8: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myInt16(const datatypes_system_cpp_pkg_interfaces::msg::Integer16 msg)
{
    // Handle myInt16 msg
    PRINT_INFO("Received myInt16: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myInt32(const datatypes_system_cpp_pkg_interfaces::msg::Integer32 msg)
{
    // Handle myInt32 msg
    PRINT_INFO("Received myInt32: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myInt64(const datatypes_system_cpp_pkg_interfaces::msg::Integer64 msg)
{
    // Handle myInt64 msg
    PRINT_INFO("Received myInt64: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myUInt8(const datatypes_system_cpp_pkg_interfaces::msg::Unsigned8 msg)
{
    // Handle myUInt8 msg
    PRINT_INFO("Received myUInt8: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myUInt16(const datatypes_system_cpp_pkg_interfaces::msg::Unsigned16 msg)
{
    // Handle myUInt16 msg
    PRINT_INFO("Received myUInt16: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myUInt32(const datatypes_system_cpp_pkg_interfaces::msg::Unsigned32 msg)
{
    // Handle myUInt32 msg
    PRINT_INFO("Received myUInt32: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myUInt64(const datatypes_system_cpp_pkg_interfaces::msg::Unsigned64 msg)
{
    // Handle myUInt64 msg
    PRINT_INFO("Received myUInt64: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myFloat32(const datatypes_system_cpp_pkg_interfaces::msg::Float32 msg)
{
    // Handle myFloat32 msg
    PRINT_INFO("Received myFloat32: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myFloat64(const datatypes_system_cpp_pkg_interfaces::msg::Float64 msg)
{
    // Handle myFloat64 msg
    PRINT_INFO("Received myFloat64: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myEnum(const datatypes_system_cpp_pkg_interfaces::msg::MyEnum msg)
{
    // Handle myEnum msg
    PRINT_INFO("Received myEnum: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myStruct(const datatypes_system_cpp_pkg_interfaces::msg::MyStructi msg)
{
    // Handle myStruct msg
    PRINT_INFO("Received myStruct: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myArray1(const datatypes_system_cpp_pkg_interfaces::msg::MyArrayOneDim msg)
{
    // Handle myArray1 msg
    PRINT_INFO("Received myArray1: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myArray2(const datatypes_system_cpp_pkg_interfaces::msg::MyArrayUnbounded msg)
{
    // Handle myArray2 msg
    PRINT_INFO("Received myArray2: %s", MESSAGE_TO_STRING(msg));
}

void consumer_consumer::handle_myArray3(const datatypes_system_cpp_pkg_interfaces::msg::MyArrayTwoDim msg)
{
    // Handle myArray3 msg
    PRINT_INFO("Received myArray3: %s", MESSAGE_TO_STRING(msg));
}

File: resources/expected/ros2/python-datatype-examples/lax/src/datatypes_system_py_pkg/datatypes_system_py_pkg/user_code/consumer_consumer_src.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from rosidl_runtime_py.convert import message_to_yaml
from datatypes_system_py_pkg.base_code.consumer_consumer_base_src import consumer_consumer_base
from datatypes_system_py_pkg_interfaces.msg import Boolean
from datatypes_system_py_pkg_interfaces.msg import Integer64
from datatypes_system_py_pkg_interfaces.msg import Float64
from datatypes_system_py_pkg_interfaces.msg import Character
from datatypes_system_py_pkg_interfaces.msg import String
from datatypes_system_py_pkg_interfaces.msg import Integer8
from datatypes_system_py_pkg_interfaces.msg import Integer16
from datatypes_system_py_pkg_interfaces.msg import Integer32
from datatypes_system_py_pkg_interfaces.msg import Unsigned8
from datatypes_system_py_pkg_interfaces.msg import Unsigned16
from datatypes_system_py_pkg_interfaces.msg import Unsigned32
from datatypes_system_py_pkg_interfaces.msg import Unsigned64
from datatypes_system_py_pkg_interfaces.msg import Float32
from datatypes_system_py_pkg_interfaces.msg import MyEnum
from datatypes_system_py_pkg_interfaces.msg import MyStructi
from datatypes_system_py_pkg_interfaces.msg import MyArrayOneDim
from datatypes_system_py_pkg_interfaces.msg import MyArrayUnbounded
from datatypes_system_py_pkg_interfaces.msg import MyArrayTwoDim
from datatypes_system_py_pkg.base_code.enum_converter import *

#===========================================================
# This file will not be overwritten when re-running Codegen
#===========================================================
class consumer_consumer(consumer_consumer_base):
    def __init__(self):
        super().__init__()
        # invoke initialize entry point
        self.initialize()

        self.get_logger().info("consumer_consumer infrastructure set up")

#=================================================
#  I n i t i a l i z e    E n t r y    P o i n t
#=================================================
    def initialize(self):
        self.get_logger().info("Initialize Entry Point invoked")

        # Initialize the node

        # Initialize the node's incoming data port values here


#=================================================
#  C o m p u t e    E n t r y    P o i n t
#=================================================
    def message_to_string(self, msg):
        yaml_str = message_to_yaml(msg)
        return yaml_str

    def handle_myBoolean(self, msg):
        # Handle myBoolean msg
        self.get_logger().info(f"Received myBoolean: {self.message_to_string(msg)}")

    def handle_myInteger(self, msg):
        # Handle myInteger msg
        self.get_logger().info(f"Received myInteger: {self.message_to_string(msg)}")

    def handle_myFloat(self, msg):
        # Handle myFloat msg
        self.get_logger().info(f"Received myFloat: {self.message_to_string(msg)}")

    def handle_myCharacter(self, msg):
        # Handle myCharacter msg
        self.get_logger().info(f"Received myCharacter: {self.message_to_string(msg)}")

    def handle_myString(self, msg):
        # Handle myString msg
        self.get_logger().info(f"Received myString: {self.message_to_string(msg)}")

    def handle_myInt8(self, msg):
        # Handle myInt8 msg
        self.get_logger().info(f"Received myInt8: {self.message_to_string(msg)}")

    def handle_myInt16(self, msg):
        # Handle myInt16 msg
        self.get_logger().info(f"Received myInt16: {self.message_to_string(msg)}")

    def handle_myInt32(self, msg):
        # Handle myInt32 msg
        self.get_logger().info(f"Received myInt32: {self.message_to_string(msg)}")

    def handle_myInt64(self, msg):
        # Handle myInt64 msg
        self.get_logger().info(f"Received myInt64: {self.message_to_string(msg)}")

    def handle_myUInt8(self, msg):
        # Handle myUInt8 msg
        self.get_logger().info(f"Received myUInt8: {self.message_to_string(msg)}")

    def handle_myUInt16(self, msg):
        # Handle myUInt16 msg
        self.get_logger().info(f"Received myUInt16: {self.message_to_string(msg)}")

    def handle_myUInt32(self, msg):
        # Handle myUInt32 msg
        self.get_logger().info(f"Received myUInt32: {self.message_to_string(msg)}")

    def handle_myUInt64(self, msg):
        # Handle myUInt64 msg
        self.get_logger().info(f"Received myUInt64: {self.message_to_string(msg)}")

    def handle_myFloat32(self, msg):
        # Handle myFloat32 msg
        self.get_logger().info(f"Received myFloat32: {self.message_to_string(msg)}")

    def handle_myFloat64(self, msg):
        # Handle myFloat64 msg
        self.get_logger().info(f"Received myFloat64: {self.message_to_string(msg)}")

    def handle_myEnum(self, msg):
        # Handle myEnum msg
        self.get_logger().info(f"Received myEnum: {self.message_to_string(msg)}")

    def handle_myStruct(self, msg):
        # Handle myStruct msg
        self.get_logger().info(f"Received myStruct: {self.message_to_string(msg)}")

    def handle_myArray1(self, msg):
        # Handle myArray1 msg
        self.get_logger().info(f"Received myArray1: {self.message_to_string(msg)}")

    def handle_myArray2(self, msg):
        # Handle myArray2 msg
        self.get_logger().info(f"Received myArray2: {self.message_to_string(msg)}")

    def handle_myArray3(self, msg):
        # Handle myArray3 msg
        self.get_logger().info(f"Received myArray3: {self.message_to_string(msg)}")


#=================================================
#  Include any additional declarations here
#=================================================
# Additions within these tags will be preserved when re-running Codegen

# Additions within these tags will be preserved when re-running Codegen
